doc/api/diagnostics_channel.md

```diff
@@ -409,6 +459,534 @@ channel.subscribe(onMessage);
+* `...args` {any} Optional arguments to pass to the function
+* Returns: {any} The return value of the given function
+
+Trace a callback-receiving function call. This will always produce a
+[`start` event][] and [`end` event][] around the synchronous portion of the
+function execution, and will produce a [`asyncStart` event][] and
+[`asyncEnd` event][] around the callback execution. It may also produce an
+[`error` event][] if the given function throws an error or the returned
+promise rejects. This will run the given function using
+[`channel.runStores(data, ...)`][] on the `start` channel.
+
+```mjs
+import diagnostics_channel from 'node:diagnostics_channel';
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.traceCallback((arg1, callback) => {
+  // Do something
+  callback(null, 'result');
+}, 1, {
+  some: 'thing',
+}, thisArg, arg1, callback);
+```
+
+```cjs
+const diagnostics_channel = require('node:diagnostics_channel');
+
+const channels = diagnostics_channel.tracingChannel('my-channel');
+
+channels.traceCallback((arg1, callback) => {
+  // Do something
+  callback(null, 'result');
+}, {
+  some: 'thing',
+}, thisArg, arg1, callback);
+```
+
+### TracingChannel Channels
+
+A TracingChannel is a collection of several channels representing specific
+points in the execution lifecycle of a single traceable action. The behaviour
+is split into five channels consisting of `start`, `end`, `asyncStart`,
+`asyncEnd`, and `error`. A single traceable action will share the same event
+object between all events, this can be helpful for managing correlation
+through a weakmap.
+
+These event objects will be extended with `result` or `error` values when
+the task "completes". In the case of a synchronous task the `result` will be
+the return value and the `error` will be anything thrown from the function.
+With callback-based async functions the `result` will be the second argument
+of the callback while the `error` will either be a thrown error visible in the
+`end` event or the first callback argument in either of the `asyncStart` or
+`asyncEnd` events.
+
+Tracing channels should follow a naming pattern of:
+
+* `tracing:module.class.method:start` or `tracing:module.function:start`
+* `tracing:module.class.method:end` or `tracing:module.function:end`
+* `tracing:module.class.method:asyncStart` or `tracing:module.function:asyncStart`
+* `tracing:module.class.method:asyncEnd` or `tracing:module.function:asyncEnd`
+* `tracing:module.class.method:error` or `tracing:module.function:error`
+
+#### `start(event)`
+
+* Name: `tracing:${name}:start`
+
+The `start` event represents the synchronous start of a traceable function.
+At this point the event data may contain function arguments or anything else
+available at the very start of the execution of the function.
+
+#### `end(event)`
+
+* Name: `tracing:${name}:end`
+
+The `end` event represents the synchronous end of a traceable function. At this
```

Comment:
Synchronous start and synchronous end could be worded more clearly to indicate that they're when a function is called and when it returns (clarifying that return does _not_ mean when `return` appears in an async function).